### PR TITLE
Implemented login date for guild members.

### PR DESCRIPTION
**Pull request by Megasantos**
Signed-off-by: Megasantos <jonataandretta@hotmail.com>

[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

This PR implements support to show the last online date of guild members.

The date used is a timestamp, this allows some modifications to the client side.

Search in msgstringtable. txt the line **3012**

**Oficial message:** `%Y.%m.%d#`

img1

As it is a timestamp I can do some cool things see

**Message 1:** `%Y.%m.%d %H:%M:%S#`

img2

**Message 2:** `%Y.%m.%d %I:%M %p#`

img3

**Message 3:** `%a, %d %b %Y %H:%M:%S#`

img4

**Message 4:** `%d %b %Y %H:%M:%S#`

img5

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1986)

**Comment by JulioCF** [comment]
Very nice!

File: src/common/mmo.h [hunk 1]
```diff
@@ -746,6 +746,7 @@ struct guild_member {
 	char name[NAME_LENGTH];
 	struct map_session_data *sd;
 	unsigned char modified;
+	uint32 login_date;
```
Comment (Asheraf) [inline src/common/mmo.h]: i would rather if you keep the same name as in char `last_login`

Comment (Megasantos) [reply src/common/mmo.h]: Done. Appveyor canceled the build of the VS 14, so error occurred

**Comment by tlacson7** [comment]
which client is supported?

**Comment by Megasantos** [comment]
@tlacson7  client > 20161026

**Comment by 4144** [comment]
can you squash last commit fix with previous commits?

File: src/char/int_guild.c [hunk 2]
```diff
@@ -1366,6 +1368,7 @@ int mapif_parse_GuildChangeMemberInfoShort(int fd, int guild_id, int account_id,
 		g->member[i].online = online;
 		g->member[i].lv = lv;
 		g->member[i].class = class;
+		g->member[i].last_login = (int)time(NULL);
```
Comment (4144) [inline src/char/int_guild.c]: last_login field in unsigned int, why you cast here to signed int?
probably replace to ``(uint32)time(NULL)``

File: src/map/guild.c [hunk 3]
```diff
@@ -275,6 +275,7 @@ void guild_makemember(struct guild_member *m,struct map_session_data *sd)
 	m->online     = 1;
 	m->position   = MAX_GUILDPOSITION-1;
 	memcpy(m->name,sd->status.name,NAME_LENGTH);
+	m->last_login = (int)time(NULL); // When player create or join a guild the date is updated
```
Comment (4144) [inline src/map/guild.c]: and here same cast

File: src/char/int_guild.c (hunk 2 above)
Comment (Megasantos) [reply src/char/int_guild.c]: I see, when I did the initial implementation I was using int, but then I went to look at the packet and noticed it was an unsigned int so I switched to uint32 and I ended up forgetting to change part of the time.

Comment (Megasantos) [reply src/char/int_guild.c]: Done.

File: src/map/guild.c (hunk 3 above)
Comment (Megasantos) [reply src/map/guild.c]: Done.

**Pull request merged**